### PR TITLE
Add Basic Script

### PR DESCRIPTION
**Pull request by SachinMeier**
parse & serialize script

to and from address

script identification

### CODE REVIEW

File: lib/script.ex [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+  def is_valid_opcode(i) when is_integer(i), do: i >= 0x00 && i < 0xFF
+
+  @doc """
+  	new returns an empty script object.
+  """
+  @spec new() :: t()
+  def new, do: %__MODULE__{items: []}
+
+  @doc """
+  	is_true? returns true if the script's only item is op_true, false otherwise.
```
Comment (bruteforcecat) [inline lib/script.ex:38]: This function is confusing to me. I think we should instead have a separate evaluate function(or even in new Evaluator module). Or there is some other use case I overlook?

File: lib/script.ex [hunk 2]
```diff
@@ -0,0 +1,669 @@
...
+      _ in MatchError -> false
+    end
+  end
+
+  @doc """
+  	is_p2wpkh? returns whether a given script is of the p2wpkh format:
+  	OP_0 OP_PUSHBYTES_20 <20-byte hash>
+  """
+  @spec is_p2wpkh?(t()) :: boolean
+  def is_p2wpkh?(script) do
```
Comment (bruteforcecat) [inline lib/script.ex:365]: https://github.com/SachinMeier/bitcoinex/pull/5

I made a simple for this. Please let me know your thought

File: lib/script.ex (hunk 1 above)
Comment (SachinMeier) [reply lib/script.ex:38]: No, you are correct. this is more for evaluate. I'll remove it.

**Review by philipglazman — APPROVED** [review]

File: lib/script.ex [hunk 3]
```diff
@@ -0,0 +1,761 @@
...
+  """
+  @spec raw_combine(t(), t()) :: t()
+  def raw_combine(%__MODULE__{items: s1}, %__MODULE__{items: s2}),
+    do: %__MODULE__{items: s1 ++ s2}
+
+  @doc """
+  	display_script returns a human readable string of the script, with
+  	op_codes shown by name rather than number. 
+  """
+  @spec display_script(t()) :: String.t()
```
Comment (bruteforcecat) [inline lib/script.ex:301]: maybe we should implement String.Char protocol using this function. I don't think we need to do this in this PR though

**Review by bruteforcecat — APPROVED** [review]

**Pull request merged**